### ISSUE TITLE
[question] how to create fullscreen window?

### ISSUE THREAD

**Issue by zhangpy** [google/gxui#94]
It is better if including such sample. thanks.

**Comment by kaly** [comment]
You can't right now. gxui is hard coded to be windowed. [drivers/gl/viewport.go#L67] (https://github.com/google/gxui/blob/master/drivers/gl/viewport.go#L67)

http://www.glfw.org/docs/latest/window.html#window_full_screen

**Comment by zhangpy** [comment]
OK. will you consider to label it as feature or enhancement?

**Comment by ben-clayton** [comment]
Sure. I suspect GLFW has good support for fullscreen windows given it is targeted at games.

**Comment by shurcooL** [comment]
It certainly does, see http://www.glfw.org/docs/latest/window.html.

**Comment by zhangpy** [comment]
I could create fullscreen by changing gxui code 
drivers/gl/viewport.go#L67
wnd, err := glfw.CreateWindow(width, height, title, nil, nil)
to 
wnd, err := glfw.CreateWindow(width, height, title, glfw.GetPrimaryMonitor(), nil)

I hope a better way, but it is OK for now, thanks :)

**Comment by ben-clayton** [comment]
Here's a better way.

**Comment by zhangpy** [comment]
Thanks to simplify it. I have to reopen it because of panic.
It will panic if I set full screen at beginning. Here is test code:
~~~
package main

import (
	"github.com/google/gxui"
	"github.com/google/gxui/drivers/gl"
	"github.com/google/gxui/samples/flags"
	"github.com/google/gxui/themes/dark"
)

var (
	theme gxui.Theme
)

func appMain(driver gxui.Driver) {
	theme = dark.CreateTheme(driver)

	window := theme.CreateWindow(400, 300, "Progress bar")
	window.SetScale(flags.DefaultScaleFactor)
	window.SetFullscreen(true)
	window.OnClose(driver.Terminate)

	label := theme.CreateLabel()
	label.SetText("test")

	layout := theme.CreateLinearLayout()
	layout.AddChild(label)

	window.AddChild(layout)
}

func main() {
	gl.StartDriver(appMain)
}
~~~
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4b1307]

goroutine 1 [running, locked to thread]:
github.com/google/gxui/drivers/gl.(*canvas).release(0x0, 0xc2081d0600)
	github.com/google/gxui/drivers/gl/canvas.go:85 +0x157
github.com/google/gxui/drivers/gl.(*canvas).Release(0x0)
	github.com/google/gxui/drivers/gl/canvas.go:249 +0x28
github.com/google/gxui/drivers/gl.func·051()
	github.com/google/gxui/drivers/gl/viewport.go:440 +0xd1
github.com/google/gxui/drivers/gl.(*driver).driverLoop(0xc208020040)
	github.com/google/gxui/drivers/gl/driver.go:82 +0x75
github.com/google/gxui/drivers/gl.StartDriver(0xc443b0)
	github.com/google/gxui/drivers/gl/driver.go:51 +0x2b9
main.main()
	main.go:40 +0x2a
-------
because panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4b1307]

goroutine 1 [running, locked to thread]:
github.com/google/gxui/drivers/gl.(*canvas).release(0x0, 0xc2081d0600)
	github.com/google/gxui/drivers/gl/canvas.go:85 +0x157
github.com/google/gxui/drivers/gl.(*canvas).Release(0x0)
	github.com/google/gxui/drivers/gl/canvas.go:249 +0x28
github.com/google/gxui/drivers/gl.func·051()
	github.com/google/gxui/drivers/gl/viewport.go:440 +0xd1
github.com/google/gxui/drivers/gl.(*driver).driverLoop(0xc208020040)
	github.com/google/gxui/drivers/gl/driver.go:82 +0x75
github.com/google/gxui/drivers/gl.StartDriver(0xc443b0)
	github.com/google/gxui/drivers/gl/driver.go:51 +0x2b9
main.main()
	main.go:40 +0x2a
-----
If I check canvas is nil in https://github.com/google/gxui/blob/master/drivers/gl/viewport.go#L439
I can avoid the panic, but I am not sure this is right way.

**Comment by ben-clayton** [comment]
It is, thanks for looking. Fixed with 120dc55ccc4e5586009dd474270d7a72b26413f8.
